Marlin/SanityCheck.h

```diff
@@ -1019,3 +1019,53 @@
 #if COUNT_LCD_24 > 1
   #error "Please select no more than one LCD controller option."
 #endif
+
+#if ENABLED( SPINDLE_ENABLE)
+  #if !PIN_EXISTS(SPINDLE_ENABLE)
+    #error "SPINDLE_ENABLE requires SPINDLE_ENABLE_PIN."
+  #elif !defined(SPINDLE_ENABLE_INVERT) || !(SPINDLE_ENABLE_INVERT == true || SPINDLE_ENABLE_INVERT == false)
+    #error "SPINDLE_ENABLE_INVERT missing or incorrectly defined"
+  #elif ENABLED (SPINDLE_SPEED) && PIN_EXISTS(SPINDLE_SPEED)
+    #if !((SPINDLE_SPEED_PIN >=2 && SPINDLE_SPEED_PIN <= 13) || (SPINDLE_SPEED_PIN >=44 && SPINDLE_SPEED_PIN <= 46) )
+      #error "SPINDLE_SPEED_PIN not assigned to a PWM pin"
+    #elif !defined(SPINDLE_SPEED_INVERT) || !(SPINDLE_SPEED_INVERT == true || SPINDLE_SPEED_INVERT == false)
+      #error "SPINDLE_SPEED_INVERT missing or incorrectly defined"
+    #elif !defined(RPM_SLOPE) || !defined(RPM_INTERCEPT) || !defined(RPM_MIN) || !defined(RPM_MAX)
+      #error "spindle speed equation constant(s) missing"
+    #elif SPINDLE_SPEED_PIN == 4 || SPINDLE_SPEED_PIN == 11 || SPINDLE_SPEED_PIN == 12 || SPINDLE_SPEED_PIN == 13
+      #error "Counter/Timer for SPINDLE_SPEED PWM pin is used by a system interrupt "
+    #elif PIN_EXISTS(X_MAX) &&  X_MAX_PIN == SPINDLE_SPEED_PIN && defined(USE_XMAX_PLUG)
+      #error "SPINDLE_SPEED pin is in use by X_MAX endstop"
+    #elif PIN_EXISTS(X_MIN) &&  X_MIN_PIN == SPINDLE_SPEED_PIN && defined(USE_XMIN_PLUG)
+      #error "SPINDLE_SPEED pin is in use by X_MIN endstop "
```

Comment:
Watch those extra spaces. If formatting isn't perfect, the PR will not be merged.